### ISSUE TITLE
`lint/style/useTemplate` should understand `.toString()` for component expressions.

### ISSUE THREAD

**Issue by lgarron** [biomejs/biome#178]
### Environment information

```block
CLI:
  Version:                      1.1.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.5.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/9.8.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Workspace:
  Open Documents:               0

Discovering running Biome servers...

[whole buncha stuff]
```

### What happened?

Biome 1.1.0 is newly warning on this for me:

```ts
const err = new Error("this is an error");
console.log("\n--------\nERROR:\n" + err.toString());
```

However, it is changing this to:

```ts
const err = new Error("this is an error");
console.log(`\n--------\nERROR:\n${err.toString()}`);
```

The `.toString()` call is always redundant in this case, since the template literal[^1] effectively invokes `.toString()` for each expression. You can verify this using the following snippet:
```js
const a = new Error();
a.toString = () => "hi";
console.log(`a: ${a}`); // Logs: "a: hi"
```

[^1]: At least when untagged. I haven't tried tagged template literals, but Biome isn't creating those either.

### Expected result

If the point of `lint/style/useTemplate` is to avoid redundant calculations, it seems to me that this should be:

```ts
const err = new Error("this is an error");
console.log(`\n--------\nERROR:\n${err}`);
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Hi! Thanks for the report!

Actually, calling `toString` on the first example is also redundant.
You can write:

```js
const err = new Error("this is an error");
console.log("\n--------\nERROR:\n" + err);
```

**Comment by lgarron** [comment]
> Actually, calling `toString` on the first example is also redundant.

Good point, that would also be nice for my linter to tell me! 😄

**Comment by Conaclos** [comment]
I see a case where `.toString()` or `String()` are required: at the start of a concatenation in order to avoid numerical addition.
For example:

```js
String(1) + 2 + "px"
```

In this case, it makes sense to remove the conversion because it becomes useless:

```diff
- String(1) + 2 + "px"
+ `${1}${2}px`
```

For uniformity, I think we could remove any useless conversion (what you proposed).

We could also add a new rule `noUselessConversion` that reports every redundant conversion (strings, numbers, and others).

**Comment by ematipico** [comment]
I am going to close this for now, because it isn't possible to achieve this for now. We need a type inference system to understand if an object implements `toString()`
